system/jlib/jutil.cpp

```diff
@@ -2663,8 +2662,79 @@ StringBuffer &getFileAccessUrl(StringBuffer &out)
     return out;
 }
 
+
+#ifdef _CONTAINERIZED
+static bool getDefaultPlane(StringBuffer &ret, const char * componentOption, const char * globalOption)
+{
+    // If the plane is specified for the component, then use that
+    if (getComponentConfigSP()->getProp(componentOption, ret))
+        return true;
+
+    //Otherwise check what the default plane for data storage is configured to be
+    if (getGlobalConfigSP()->getProp(globalOption, ret))
+        return true;
+
+    return false;
+}
+
+static bool getDefaultPlaneDirectory(StringBuffer &ret, const char * componentOption, const char * globalOption)
+{
+    StringBuffer planeName;
+    if (!getDefaultPlane(planeName, componentOption, globalOption))
+        return false;
+
+    Owned<IPropertyTree> storagePlane = getStoragePlane(planeName);
+    return storagePlane->getProp("@prefix", ret);
+}
+#endif
+
 bool getConfigurationDirectory(const IPropertyTree *useTree, const char *category, const char *component, const char *instance, StringBuffer &dirout)
 {
+#ifdef _CONTAINERIZED
+    if (streq(category, "data"))
```

Comment:
Perhaps assert that useTree is null?

Reply:
It is non null in too many places at the moment - possibly once all the calls have been cleaned up.